src/main.ts

```diff
@@ -1,8 +1,9 @@
 import CLI from './cli';
 import SpaceCraft from './spacecraft';
 import { Earth, Moon } from './locations';
+import SpacecraftEmitter from './spacecraftEmitter';
 
-const spacecraft = new SpaceCraft(Earth, Moon);
+const spacecraft = new SpaceCraft(Earth, Moon, new SpacecraftEmitter(process.stdout));
```

Comment:
I think you could declare the SpacecraftEmitter in a const and then, pass the instance to the SpaceCraft:

```suggestion
const spaceCraftEmitter = new SpacecraftEmitter(process.stdout);
const spacecraft = new SpaceCraft(Earth, Moon, spaceCraftEmitter);
```